src/shim/gas.rs

```diff
@@ -77,7 +79,8 @@ impl From<GasV3> for Gas {
     }
 }
 
-pub struct GasCharge(GasChargeV3);
+#[derive(Debug)]
+pub struct GasCharge(pub GasChargeV3);
```

Comment:
The point of this structure is to hide the `GasChargeV3` type. All queries should go through the shim.

```suggestion
pub struct GasCharge(GasChargeV3);
```